llvm/include/llvm/ProfileData/InstrProfCorrelator.h

```diff
@@ -35,20 +36,32 @@ class InstrProfCorrelator {
   /// correlate.
   enum ProfCorrelatorKind { NONE, DEBUG_INFO, BINARY };
 
+  struct WarningCounter {
```

Comment:
Could the WarningCounter refactoring be split out and landed separately?

Reply:
Since there is [debate](https://github.com/llvm/llvm-project/pull/75957#discussion_r1433009581) about if this change is necessary at all, let's keep the change here in case we decided it's unnecessary.